geoh5py/shared/concatenation/drillholes_group_table.py

```diff
@@ -151,19 +153,22 @@ def _get_property_groups(parent, name) -> dict[UUID, ConcatenatedPropertyGroup]:
         return property_groups
 
     @staticmethod
-    def _pad_arrays_to_first(arrays: list[np.ndarray], ndv: Any) -> list[np.ndarray]:
+    def _pad_arrays_to_first(
```

Comment:
Feels like this shouldn't be static, and be called something more explicit about what it does.
Maybe more like `def pad_array_to_association(self)`? The table should know about its association lenght.

Reply:
I agree for the name. 

For the static vs non static, it doesn't import much. 
I'll call index by drillhole if you consider it important.

Reply:
Well if it's not used externally, and relies on the state of local attributes (i.e. `association`) then it makes more sense to be a regular method.